src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,135 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private static User user;
+    private static RegistrationServiceImpl registrationService;
+
+    @BeforeAll
+    static void beforeAll() {
+        registrationService = new RegistrationServiceImpl();
+    }
+
+    @BeforeEach
+    void beforeEach() {
+        user = new User();
+        user.setLogin("potrap");
+        user.setPassword("qwerty123");
+        user.setAge(20);
+    }
+
+    @AfterEach
+    void afterEach() {
+        Storage.people.clear();
+    }
+
+    @Test
+    void register_emptyLogin_notOk() {
+        user.setLogin(" ");
+        assertThrows(InvalidInputDataException.class, () -> {
+            registrationService.register(user);
+        }, "InvalidRegistrationDataException expected for login [ ]");
+    }
+
+    @Test
+    void register_emptyPassword_notOk() {
+        user.setPassword(" ");
+        assertThrows(InvalidInputDataException.class, () -> {
+            registrationService.register(user);
+        }, "InvalidRegistrationDataException expected for password [ ]");
+    }
+
+    @Test
+    void register_nullAge_notOk() {
+        user.setAge(null);
+        assertThrows(InvalidInputDataException.class, () -> {
+            registrationService.register(user);
+        }, "InvalidRegistrationDataException expected for age null");
+    }
+
+    @Test
+    void register_ageLessThanEighteen_notOk() {
+        user.setAge(14);
```

Comment:
for testing it is better to take marginally unacceptable input data, in this case 17 years